File: app/src/main/java/com/example/yuwathi/models/Complaint.java
package com.example.yuwathi.models;

import java.util.ArrayList;
import java.util.List;

public class Complaint {
    private String id;
    private String userId;
    private String title;
    private String location;
    private String date;
    private String status;
    private String priority;
    private String description;
    private String witnesses;
    private String vehicle;
    private String suspectDescription;
    private boolean ongoing;
    private String contactPreference;
    private String evidence;
    private List<String> evidenceUrls = new ArrayList<>();

    public Complaint() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getPriority() {
        return priority;
    }

    public void setPriority(String priority) {
        this.priority = priority;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getWitnesses() {
        return witnesses;
    }

    public void setWitnesses(String witnesses) {
        this.witnesses = witnesses;
    }

    public String getVehicle() {
        return vehicle;
    }

    public void setVehicle(String vehicle) {
        this.vehicle = vehicle;
    }

    public String getSuspectDescription() {
        return suspectDescription;
    }

    public void setSuspectDescription(String suspectDescription) {
        this.suspectDescription = suspectDescription;
    }

    public boolean isOngoing() {
        return ongoing;
    }

    public void setOngoing(boolean ongoing) {
        this.ongoing = ongoing;
    }

    public String getContactPreference() {
        return contactPreference;
    }

    public void setContactPreference(String contactPreference) {
        this.contactPreference = contactPreference;
    }

    public String getEvidence() {
        return evidence;
    }

    public void setEvidence(String evidence) {
        this.evidence = evidence;
    }

    public List<String> getEvidenceUrls() {
        return evidenceUrls;
    }

    public void setEvidenceUrls(List<String> evidenceUrls) {
        this.evidenceUrls = evidenceUrls;
    }
}

File: app/src/main/java/com/example/yuwathi/models/Complaint.kt
package com.example.yuwathi.models

import java.util.Date

data class Complaint(
    val id: String = "",
    val userId: String = "",
    val userName: String = "",
    val title: String = "",
    val description: String = "",
    val category: ComplaintCategory = ComplaintCategory.OTHER,
    val urgency: UrgencyLevel = UrgencyLevel.MEDIUM,
    val status: ComplaintStatus = ComplaintStatus.PENDING,
    val timestamp: Date = Date(),
    val evidenceUrls: List<String> = emptyList(),
    val location: String = "",
    val resolvedBy: String = "",
    val resolvedDate: Date? = null,
    val adminNotes: String = ""
)

enum class ComplaintCategory {
    HARASSMENT,
    STALKING,
    ASSAULT,
    CYBERBULLYING,
    DOMESTIC_VIOLENCE,
    OTHER
}

enum class UrgencyLevel {
    LOW,
    MEDIUM,
    HIGH,
    CRITICAL
}

enum class ComplaintStatus {
    PENDING,
    UNDER_REVIEW,
    RESOLVED,
    DISMISSED,
    ESCALATED
}
